github4s/src/main/scala/github4s/algebras/Issues.scala

```diff
@@ -14,19 +14,12 @@
  * limitations under the License.
  */
 
-package github4s.api
+package github4s.algebras
 
 import github4s.GithubResponses.GHResponse
-import github4s.{GithubApiUrls, HttpClient, HttpRequestBuilderExtension}
-import github4s.free.domain._
-import github4s.util.URLEncoder
-import io.circe.syntax._
-import io.circe.generic.auto._
+import github4s.domain._
 
-/** Factory to encapsulate calls related to Issues operations  */
-class Issues[M[_]](implicit urls: GithubApiUrls, httpClientImpl: HttpRequestBuilderExtension[M]) {
-
-  val httpClient = new HttpClient[M]
+trait Issues[F[_]] {
```

Comment:
Out of curiosity, do we actually need these traits? Does it make sense if we directly define the algebra along with the implementation without needing to override these "interfaces"? Cannot think in any use case where these traits might be useful 🤔

Reply:
Functionally, that makes sense. The current implementation follows a tagless final pattern, thus separating the algebras and interpreters, but it only has one interpretation. 

I only have 2 weak arguments to keep the current definitions.
1. To follow tagless final pattern clearly
2. To cater to some other implementations in the future.

Idk what other interpreters would be interesting... 

- Maybe offering more than one choice of http client?

- support a Github V3 interface for the Github V4 API...
🤷‍♂ 

What do you think @rafaparadela ?

Reply:
One of the purposes of this PR is addressing the migration from _Free_ to _Tagless Final_, which normally expresses the algebras as type classes. And the modules as a composition of some of these algebras. 
The fact of having algebras as type classes allow generally to write different interpreters for runtime or testing. Another argument not applicable in Github4s, could be the capability to rewrite another interpretation of the algebra (using different HTTP client, or different DB driver, or whatever change) and having both of the co-existing, and respecting the trait as a contract).

I don't really see either a solid argument in favor of having these traits, beyond the fact of respecting the pattern and keep using this project a didactic example of it.

Reply:
Thanks both! We can move on as is, a lot of work here :)

I already approved, so LGTM!